app/src/controllers/submission.ts

```diff
@@ -0,0 +1,175 @@
+import config from 'config';
+import { NIL, v4 as uuidv4 } from 'uuid';
+
+import { APPLICATION_STATUS_LIST } from '../components/constants';
+import { getCurrentIdentity, isTruthy, toTitleCase } from '../components/utils';
+import { submissionService, permitService, userService } from '../services';
+
+import type { NextFunction, Request, Response } from '../interfaces/IExpress';
+import type { ChefsFormConfig, ChefsFormConfigData, Submission, ChefsSubmissionExport, Permit } from '../types';
+
+const controller = {
+  checkAndStoreNewSubmissions: async () => {
+    const cfg = config.get('server.chefs.forms') as ChefsFormConfig;
+
+    const permitTypes = await permitService.getPermitTypes();
+
+    // Mapping of SHAS intake permit names to PCNS types
+    const shasPermitMapping = new Map<string, string>([
+      ['archaeologySiteAlterationPermit', 'Alteration'],
+      ['archaeologyHeritageInspectionPermit', 'Inspection'],
+      ['archaeologyInvestigationPermit', 'Investigation'],
+      ['forestsPrivateTimberMark', 'Private Timber Mark'],
+      ['forestsOccupantLicenceToCut', 'Occupant Licence To Cut'],
+      ['landsCrownLandTenure', 'Commercial General'],
+      ['roadwaysHighwayUsePermit', 'Highway Use Permit'],
+      ['siteRemediation', 'Contaminated Sites Remediation'],
+      ['subdividingLandOutsideAMunicipality', 'Rural Subdivision'],
+      ['waterChangeApprovalForWorkInAndAboutAStream', 'Change Approval for Work in and About a Stream'],
+      ['waterLicence', 'Water Licence'],
+      ['waterNotificationOfAuthorizedChangesInAndAboutAStream', 'Notification'],
+      ['waterShortTermUseApproval', 'Use Approval'],
+      ['waterRiparianAreasProtection', 'New'],
+      ['waterRiparianAreasProtection', 'New']
+    ]);
+
+    const exportData: Array<Partial<Submission & { activityId: string; formId: string; permits: Array<Permit> }>> =
+      await Promise.all(
```

Comment:
Not completely certain looking at this, but if you're doing a Promise.all(...).then(...) you don't need an await to do an unwrapped assignment.